demos/demo.cpp

```diff
@@ -152,17 +158,20 @@ int main()
   Action action;
   Observation observation;
 
-  //robot_interfaces::TimeIndex index = frontend.get_current_timeindex();
-  robot_interfaces::TimeIndex index = 0;
+  // simulated action :
+  // 1 dof going from 200 to 300
+  // The other going from 300 to 200
   
   for(uint value=200;value<=300;value++)
     {
       action.values[0]=value;
       action.values[1]=500-value;
-      frontend.append_desired_action(action);
+      // this action will be stored at index
+      robot_interfaces::TimeIndex index = frontend.append_desired_action(action);
       frontend.wait_until_timeindex(index+1);
```

Comment:
This wait is not needed (`get_observation` will already wait internally).

Reply:
removed !